contract-sdk/specs/oas20/README.md

```diff
@@ -0,0 +1,83 @@
+
+[ERC20]: https://ethereum.org/en/developers/docs/standards/tokens/erc-20/
+[CW20]: https://github.com/CosmWasm/cw-plus/blob/main/packages/cw20/README.md
+
+### Requests
+
+`Instantiate(TokenInstantiation)` - Instantiates the OAS20 token contract.
+
+```rust
+/// OAS20 token instantiation information.
+#[derive(Debug, Default, Clone, cbor::Decode, cbor::Encode, PartialEq, Eq)]
+pub struct TokenInstantiation {
+    /// Name of the token.
+    pub name: String,
+    /// Token symbol.
+    pub symbol: String,
+    /// Number of decimals.
+    pub decimals: u8,
+    /// Initial balances of the token.
+    #[cbor(optional, default, skip_serializing_if = "Vec::is_empty")]
+    pub initial_balances: Vec<InitialBalance>,
+    /// Information about minting in case the token is mintable.
+    #[cbor(optional)]
+    pub mintable: Option<MintableInformation>,
+}
+```
+
+`Transfer{ to: Address, amount: u128 }` - Moves `amount` of tokens from `ctx.caller_address()` to the `to` account.
+
+`Send { to: InstanceId, amount: u128, msg: cbor::Value }` - Moves `amount` of tokens from `ctx.caller_address()` to the `to` contract and calls `ReceiveOas20` on the receiver contract. `to` is an instance ID of the receiving contract (not its address, to prevent sending to a non-contract address). The receiving contract should handle the `Receive { sender: Address, amount: u128, msg: cbor::Value }` call.
+
+`Burn { amount: u128 }` - Removes `amount` of tokens from `ctx.caller_address()` and reduces the `total_supply` by `amount`.
+
+### Queries
+
+`TokenInformation` - Returns the general token information. Response:
+
+```rust
+/// OAS20 token information.
+#[derive(Debug, Clone, cbor::Decode, cbor::Encode, PartialEq, Eq)]
+pub struct TokenInformation {
+    /// Name of the token.
+    pub name: String,
+    /// Token symbol.
+    pub symbol: String,
+    /// Number of decimals.
+    pub decimals: u8,
+    /// Total supply of the token.
+    pub total_supply: u128,
+    /// Information about minting in case the token is mintable.
+    #[cbor(optional)]
+    pub mintable: Option<MintableInformation>,
+}
+```
+
+`Balance { account: Address }` - Returns the balance of the account.
+
+## Mintable (optional)
+
+Tokens supporting to be minted (potentially with a cap). Only a single minter is supported - for a use case with multiple minters a multi-sig minter address should be used.
+
+### Requests
+
+`Mint { to: Address, amount: u128 }` - Adds `amount` of tokens to `to` account and increase the `total_supply` bt `amount`. Only the `minter` is allowed to issue this.
+
+### Queries
+
+`MinterInfo` - Returns the minter address and the minting cap.
+
+```rust
+/// Token mintable information.
+#[derive(Debug, Default, Clone, cbor::Decode, cbor::Encode, PartialEq, Eq)]
+pub struct MintableInformation {
+    /// Caller address which is allowed to mint new tokens.
+    pub minter: Address,
+    /// Maximum number of minted tokens.
```

Comment:
I checked the sample implementation, and the cap is a limit for the total supply. that might be subtly different from 'number of minted tokens' when (i) the token starts out with some nonzero supply and/or (ii) someone burns tokens

Reply:
That's true, i'll update the docs, since cap being the total supply limit is what makes more sense. It's also what related specs do (CW20).